store/schedule.go

```diff
@@ -268,40 +310,57 @@ func (s *Schedule) UpdateReconciliationHistory(status Status, errMsg string) {
 }
 
 func (s *Schedule) UnmarshalJSON(data []byte) error {
+	// Define an auxiliary type to prevent recursive calls to UnmarshalJSON
 	type Alias Schedule
 	aux := &struct {
 		*Alias
+		HttpCallbackData   *HTTPCallback   `json:"httpCallback,omitempty"`
+		AirbusCallbackData *AirbusCallback `json:"airbusCallback,omitempty"`
 	}{
 		Alias: (*Alias)(s),
 	}
+
+	// Unmarshal into the auxiliary struct
 	if err := json.Unmarshal(data, &aux); err != nil {
 		return err
 	}
 
-	var callbackData struct {
-		Type string `json:"type"`
-	}
+	// Check if CallbackRaw is present and use specific logic
+	if len(s.CallbackRaw) > 0 {
+		var callbackData struct {
+			Type string `json:"type"`
+		}
 
-	if err := json.Unmarshal(s.CallbackRaw, &callbackData); err != nil {
-		return err
-	}
+		if err := json.Unmarshal(s.CallbackRaw, &callbackData); err != nil {
+			return err
+		}
 
-	factoryFunc, ok := Registry[callbackData.Type]
-	if !ok {
-		return fmt.Errorf("unknown callback type: %s", callbackData.Type)
-	}
+		factoryFunc, ok := Registry[callbackData.Type]
+		if !ok {
+			return fmt.Errorf("unknown callback type: %s", callbackData.Type)
+		}
 
-	callback := factoryFunc()
-	if err := json.Unmarshal(s.CallbackRaw, callback); err != nil {
-		return err
+		callback := factoryFunc()
+		if err := json.Unmarshal(s.CallbackRaw, callback); err != nil {
+			return err
+		}
+
+		s.Callback = callback
+	} else {
+		// Use the HttpCallbackData or AirbusCallbackData
+		if aux.HttpCallbackData != nil {
+			s.HttpCallback = *aux.HttpCallbackData
+		}
+		if aux.AirbusCallbackData != nil {
+			s.AirbusCallback = *aux.AirbusCallbackData
+		}
 	}
 
-	s.Callback = callback
 	return nil
 }
 
-func (s *Schedule) ValidateSchedule() []string {
-	glog.Infof("ValidateSchedule: %+v", s)
+func (s *Schedule) ValidateSchedule(app App, conf conf.AppLevelConfiguration) []string {
+	glog.V(constants.INFO).Infof("ValidateSchedule: %+v", s)
```

Comment:
why there is need to add .V(constants.INFO) even after using .Infof?

Reply:
We are using V-style logging from glog to control the logging levels.
https://pkg.go.dev/github.com/golang/glog